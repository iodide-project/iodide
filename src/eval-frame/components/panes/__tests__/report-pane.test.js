import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";

import ReportPane from "../report-pane";

const mockStore = configureStore([]);

describe("ReportPane", () => {
  [
    { name: "empty", reportChunks: [] },
    {
      name: "single markdown chunk",
      reportChunks: [
        { chunkId: "1", chunkType: "md", chunkContent: "*cheezburger*" }
      ]
    },
    {
      name: "single markdown chunk, css chunk",
      reportChunks: [
        { chunkId: "1", chunkType: "md", chunkContent: "*cheezburger*" },
        {
          chunkId: "2",
          chunkType: "css",
          chunkContent:
            "#htmlInMd {border: 2px solid gray; margin: 8px; padding: 4px;}"
        }
      ]
    },
    {
      name: "single markdown chunk, other kind of chunk",
      reportChunks: [
        { chunkId: "1", chunkType: "md", chunkContent: "*cheezburger*" },
        { chunkId: "2", chunkType: "mmd", chunkContent: "MERMAIDFTW" }
      ]
    }
  ].forEach(testCase => {
    test(`report pane: ${testCase.name}`, () => {
      const store = mockStore({
        reportChunks: testCase.reportChunks
      });

      const component = renderer.create(
        <Provider store={store}>
          <ReportPane />
        </Provider>
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
