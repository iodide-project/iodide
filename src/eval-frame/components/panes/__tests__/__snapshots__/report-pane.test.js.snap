// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportPane report pane: empty 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "overflow": "auto",
    }
  }
/>
`;

exports[`ReportPane report pane: single markdown chunk 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "overflow": "auto",
    }
  }
>
  <div
    className="user-markdown"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p><em>cheezburger</em></p>
",
      }
    }
  />
</div>
`;

exports[`ReportPane report pane: single markdown chunk, css chunk 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "overflow": "auto",
    }
  }
>
  <div
    className="user-markdown"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p><em>cheezburger</em></p>
",
      }
    }
  />
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "#htmlInMd {border: 2px solid gray; margin: 8px; padding: 4px;}",
      }
    }
  />
</div>
`;

exports[`ReportPane report pane: single markdown chunk, other kind of chunk 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "overflow": "auto",
    }
  }
>
  <div
    className="user-markdown"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p><em>cheezburger</em></p>
",
      }
    }
  />
  <div
    id="side-effect-target-2"
  />
</div>
`;
